src/osm_configurator/model/project/calculation/paralellization/work_manager.py

```diff
@@ -0,0 +1,54 @@
+from __future__ import annotations
+
+import multiprocessing
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from typing import Any
+    from typing import List
+    from typing import Tuple
+    from typing import Callable
+    from src.osm_configurator.model.project.calculation.paralellization.work import Work
+    from multiprocessing import Pool
+
+
+class WorkManager:
+    """
+    The WorkManager allow a parallelization of work via multiprocessing.
+    """
+    def __init__(self, max_processes: int):
+        """
+        Creates an empty work manager.
+        Args:
+            max_processes (int): The maximal number of processes that might be run at the same time
+        """
+        self._max_workers = max_processes
+        self._work_to_do: List[Work] = []
+
+    def append_work(self, work: Work):
+        """
+        Adds a piece of work to the manager
+        """
+        self._work_to_do.append(work)
+
+    def do_all_work(self) -> List:
+        """
+        Executes all work that has been append till now.
+
+        Returns:
+            List: A list of the return-values of all the work.
+        """
+        pool: Pool
+        with multiprocessing.Pool(processes=self._max_workers) as pool:
+            return pool.map(self._worker_entry_point, self._work_to_do)
```

Comment:
we might want to use imap instead of map

https://stackoverflow.com/questions/26520781/multiprocessing-pool-whats-the-difference-between-map-async-and-imap

![Screenshot from 2023-02-10 00-04-20](https://user-images.githubusercontent.com/36456825/217958807-f448fa38-44f8-432a-b973-7f4811fc8e3e.png)

Reply:
maybe test it idk.

Reply:
we already have a list of the traffic cells. So this isn't really a problem. If we would want to use imap, we would need to converrt that list into a itterator, but I don't see the benefit in that.